esp/services/ws_sql/ws_sqlService.cpp

```diff
@@ -1219,8 +1215,7 @@ bool CwssqlEx::onExecutePreparedSQL(IEspContext &context, IEspExecutePreparedSQL
 {
    try
    {
-       if (!context.validateFeatureAccess(WSSQLACCESS, SecAccess_Write, false))
-           throw MakeStringException(-1, "Failed to execute SQL. Permission denied.");
+       context.ensureFeatureAccess(WSSQLACCESS, SecAccess_Write, -1, "Failed to execute SQL. Permission denied.");
```

Comment:
original error message was wrong. should be "Failed to execute Prepared SQL. Permission denied."